themes/powerline-plain/powerline-plain.theme.bash

```diff
@@ -56,3 +56,5 @@ POWERLINE_PROMPT=${POWERLINE_PROMPT:="user_info scm python_venv ruby cwd"}
 
 safe_append_prompt_command __powerline_prompt_command
 
+
+HISTORY_AUTOSAVE=${HISTORY_AUTOSAVE:-0}
```

Comment:
A couple of comments:

* You should probably define the variable a couple of lines up, with the other variables - before the `safe_append_prompt_command` function is called. Just to make sure that the variable is defined before it is used for the first time.
* To make it reusable across more themes, maybe move the variable and its usage either to the `base` theme, or to the `powerline` theme - this way, it's easier to reuse. The `powerline-plain` theme that you added it to is including the `powerline` theme - this should also work...

Reply:
Hi there, 

I've had a chance to make requested changes and seems it's working smoothly and now it's better I guess 

Thanks for the heads up.

Reply:
I think it's better if we add a function such as `save_history_immediately`, it could be better instead every theme is going to have the very same line of code. 
I couldn't decide where to put that honestly. If anyone can point that out, I can add that as well.

Reply:
If you want to create a function that is available to all themes, you can place it in the `base.theme.bash` file: https://github.com/Bash-it/bash-it/blob/master/themes/base.theme.bash

Reply:
Just noticed that you already moved it there... Sorry.